pkg/api/auth/public_api_handler.go

```diff
@@ -0,0 +1,146 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/rancher/harvester/pkg/config"
+	ctlcorev1 "github.com/rancher/wrangler-api/pkg/generated/controllers/core/v1"
+
+	dashboardauthapi "github.com/kubernetes/dashboard/src/app/backend/auth/api"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/rest"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+)
+
+const (
+	//action
+	actionQuery         = "action"
+	loginActionName     = "login"
+	skippableActionName = "skippable"
```

Comment:
`skippable` is redundant, I think we should be able to use the login action directly.